webapp/pages/profile/_id/_slug.vue

```diff
@@ -348,6 +379,19 @@ export default {
         return []
       }
       return this.uniq(this.user.contributions.filter(post => !post.deleted))
+    },
+    socialMediaLinks() {
+      const { socialMedia = [] } = this.user
+      return socialMedia.map(socialMedia => {
+        const { url } = socialMedia
+        const matches = url.match(
+          /^(?:https?:\/\/)?(?:[^@\n])?(?:www\.)?([^:\/\n?]+)/g
```

Comment:
I wonder if we could avoid to roll our own URL regex. How about using NodeJS `url` module? https://stackoverflow.com/questions/30931079/validating-a-url-in-node-js

Reply:
that is an interesting idea.. I'm ok to do that, I just don't know how other people feel about adding an npm package for just that, but maybe I am reading a different answer on that SO post you posted :point_up:

Reply:
[nodejs url](https://nodejs.org/api/url.html) is a package which is available by default on NodeJS. Maybe we have to polyfill it on the browser (if it's not already polyfilled by Vue&friends).